src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -69,24 +71,29 @@ void CPManager::register_consumer(cp_consumer_t consumer_id, std::unique_ptr< CP
     }
 }
 
+[[nodiscard]] CPGuard CPManager::cp_guard() { return CPGuard{}; }
+
 CP* CPManager::cp_io_enter() {
     rcu_read_lock();
     auto cp = get_cur_cp();
     if (!cp) {
         rcu_read_unlock();
```

Comment:
When we fall into this if, we return nullptr to caller, which is in this PR will be `CPGuard::CPGuard`, is it expected that we will be pushing nullptr to the stack?

Reply:
It is ok for us to push nullptr, because its destructor checks if cp is nullptr (exactly to cover this case).